[APM] Transaction duration anomaly alerting integration

ogupte opened the pull request:
Closes #72636.

Adds alerting integration for APM transaction duration anomalies.

![Screen Shot 2020-08-26 at 5 18 04 PM](https://user-images.githubusercontent.com/1967266/91369243-280e8a00-e7c0-11ea-9b30-deb7da1c45b5.png)

ogupte committed:
Closes #72636. Adds alerting integration for APM transaction duration anomalies.

formgeist commented:
@ogupte Looking good - the only thing that stumped me was that we're using the actual score in the summary. I think it would easier to understand if we simply showed the levels, since you're not able to specify a specific anomaly score value, but only pick the severity level.

<img width="372" alt="Screenshot 2020-08-24 at 10 42 58" src="https://user-images.githubusercontent.com/4104278/91023680-25146d80-e5f7-11ea-95b0-b8c2ad2ee2cd.png">

I think @nehaduggal had this in her screenshot;

![image](https://user-images.githubusercontent.com/4104278/91023699-2ba2e500-e5f7-11ea-9df7-99c0e512dc70.png)

Any challenges with doing this?

sqren commented on x-pack/plugins/apm/public/components/app/ServiceDetails/index.tsx:
```diff
@@ -26,19 +26,18 @@ export function ServiceDetails({ tab }: Props) {
-
-  const { core } = useApmPluginContext();
+  const isMlPluginEnabled = 'ml' in plugin.plugins;
+  const canReadAnomalies = !!(
+    isMlPluginEnabled &&
+    capabilities.ml.canAccessML &&
+    capabilities.ml.canGetJobs
+  );
```
nit: looks like all three are booleans so there should be no reason to coerce with `!!`

ogupte replied:
I added the double negative because the logical and expression returns the type of the last operand (`capabilities.ml.canGetJobs`) which is: `boolean | Readonly<{[x: string]: boolean;}>`. Without the double NOT operator, `canReadAnomalies` fails the type assignment for the `canReadAnomalies` prop of the `AlertIntegrations` component.

sqren commented on x-pack/plugins/apm/public/components/shared/TransactionDurationAnomalyAlertTrigger/SelectAnomalySeverity.tsx:
```diff
@@ -0,0 +1,113 @@
+  const options = ANOMALY_SCORES.map((anomalyScore) =>
+    getOption(theme, anomalyScore)
+  );
+  const [anomalyScore, setAnomalyScore] = useState<SeverityScore>(value);
+
+  useEffect(() => {
+    setAnomalyScore(value);
+  }, [value]);
```
Thought: Either this component should be stateless or `value` should be called `initialValue`

sqren commented on x-pack/plugins/apm/public/components/shared/TransactionDurationAnomalyAlertTrigger/index.tsx:
```diff
@@ -0,0 +1,153 @@
+      }
+    />,
+    <PopoverExpression
+      value={
+        params.environment === ALL_OPTION.value
+          ? ALL_OPTION.text
+          : params.environment
+      }
```
Can we use `[getEnvironmentLabel](https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/common/environment_filter_values.ts#L12)`?

```suggestion
      value={getEnvironmentLabel(params.environment)}
```

sqren replied:
Btw this is repeated in 

https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/components/shared/ErrorRateAlertTrigger/index.tsx#L54-L58

and 

https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/components/shared/TransactionDurationAlertTrigger/index.tsx#L67-L71

Might make sense to update there as well.

sqren replied:
While you are at it:
can you move [`ALL_OPTION`](https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/hooks/useEnvironments.tsx#L16-L21) and [`NOT_DEFINED_OPTION`](https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/components/shared/EnvironmentFilter/index.tsx#L35-L40) to 

https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/common/environment_filter_values.ts#L8-L14 ?

sqren commented on x-pack/plugins/apm/public/components/shared/TransactionDurationAnomalyAlertTrigger/index.tsx:
```diff
@@ -0,0 +1,153 @@
+        onChange={(e) =>
+          setAlertParams(
+            'transactionType',
+            e.target.value as Params['transactionType']
```
Would be nice to avoid `as Params['transactionType']`. Is it because `e.target.value` can be other types than `string`?

ogupte replied:
We can probably safely avoid the type assertion. It's a holdover from [`TransactionDurationAlertTrigger`](https://github.com/elastic/kibana/blob/ec08dcae074a283b6be7877fa177bf8ab2b66db9/x-pack/plugins/apm/public/components/shared/TransactionDurationAlertTrigger/index.tsx#L105) which this component was based on.

sqren commented on x-pack/plugins/apm/server/lib/alerts/register_transaction_duration_anomaly_alert_type.ts:
```diff
@@ -0,0 +1,136 @@
+      }
+      const alertParams = params as TypeOf<typeof paramsSchema>;
+      const mlClient = services.getLegacyScopedClusterClient(ml.mlClient);
+      const request = { params: 'DummyKibanaRequest' } as KibanaRequest;
```
Is "DummyKibanaRequest" some leftover boiletplate?

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts:
```diff
@@ -136,16 +138,19 @@ function transformResponseToServiceAnomalies(
 
 export async function getMLJobIds(
-  ml: Required<Setup>['ml'],
+  ml: {
+    anomalyDetectors: ReturnType<MlPluginSetup['anomalyDetectorsProvider']>;
+  },
```
Perhaps just change the signature:


```suggestion
  anomalyDetectors: Required<Setup>['ml']['anomalyDetectors'],
```

sqren commented on x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts:
```diff
@@ -136,16 +138,19 @@ function transformResponseToServiceAnomalies(
     (job) => (job.custom_settings?.job_tags?.apm_ml_version ?? 0) >= 2
   );
-  if (environment) {
+  if (environment && environment !== ENVIRONMENT_ALL) {
```
Was it a bug that we didn't take this into account previously?
also: this would go away if we got rid of "all" envs.

ogupte replied:
This wasn't a bug before since the environment value was omitted (in the `uiFilters` object) when "All" was the selected environment filter. If we get rid of the "All" as a environment filter option from the alert creation overlay, then we can get rid of this. We could also omit this check if we do this check in the alert executor function up the call stack.

sqren commented on x-pack/plugins/ml/server/shared_services/providers/anomaly_detectors.ts:
```diff
@@ -23,7 +23,13 @@ export function getAnomalyDetectorsProvider({
   return {
     anomalyDetectorsProvider(mlClusterClient: ILegacyScopedClusterClient, request: KibanaRequest) {
-      const hasMlCapabilities = getHasMlCapabilities(request);
+      //  APM is using this service in anomaly alert, kibana alerting doesn't provide request object
+      // So we are adding a dummy request for now
+      // TODO: Remove this once kibana alerting provides request object
+      const hasMlCapabilities =
+        request.params !== 'DummyKibanaRequest'
+          ? getHasMlCapabilities(request)
+          : (_caps: string[]) => Promise.resolve();
```
Okay, I see now why you used "DummyKibanaRequest". This doesn't seem like the right approach. Either `request` should be optional or we should not use `anomalyDetectorsProvider`. Passing in a dummy request object is going to cause confusion down the line I think.

jgowdyelastic replied:
`DummyKibanaRequest` will not be needed after this PR is merged https://github.com/elastic/kibana/pull/74965
However a fake request object will still be needed until https://github.com/elastic/kibana/issues/64588 is fixed.
Making `request` optional will require more changes to undo our work arounds once this issue is resolved.

ogupte replied:
@jgowdyelastic do i need to hold off on merging this PR until yours is merged first? LMK if i should make any changes to the `DummyKibanaRequest` in the meantime.

jgowdyelastic replied:
that PR is a few days away from being merged. so i'm happy incorporate and update the changes in this PR if you merge first.

ogupte committed:
Merge remote-tracking branch 'upstream/master' into apm-72636-anomaly-detection-based-alerts

ogupte committed:
Code review feedback

ogupte committed:
Display alert summary with the selected anomaly severity label instead of the anomaly score.

ogupte committed:
- refactored ALL_OPTION and NOT_DEFINED_OPTION to be shared from common/environment_filter_values
- utilize getEnvironmentLabel in the alerting trigger components and added support for the 'All' label

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/)
* Commit: 94c3b48fb0f38921c0c0449c470c0adf52402ce2
* This comment will update when the build is complete

### Failed CI Steps
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/execution/node/471/log)
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/execution/node/582/log)

### History
* :green_heart: [Build #69862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/) succeeded ec08dcae074a283b6be7877fa177bf8ab2b66db9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/","number":"69862","commit":"ec08dcae074a283b6be7877fa177bf8ab2b66db9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/","number":"70444","commit":"94c3b48fb0f38921c0c0449c470c0adf52402ce2"}]}
PIPELINE-->

sqren commented on x-pack/plugins/apm/common/environment_filter_values.ts:
```diff
@@ -9,11 +9,29 @@ import { i18n } from '@kbn/i18n';
 export const ENVIRONMENT_ALL = 'ENVIRONMENT_ALL';
 export const ENVIRONMENT_NOT_DEFINED = 'ENVIRONMENT_NOT_DEFINED';
 
+const ENVIRONMENT_ALL_LABEL = i18n.translate('xpack.apm.environment.allLabel', {
+  defaultMessage: 'All',
+});
+const ENVIRONMENT_NOT_DEFINED_LABEL = i18n.translate(
+  'xpack.apm.filter.environment.notDefinedLabel',
+  { defaultMessage: 'Not defined' }
+);
```
Should the translation keys be aligned?

sqren commented on x-pack/plugins/apm/common/environment_filter_values.ts:
```diff
@@ -9,11 +9,29 @@ import { i18n } from '@kbn/i18n';
 export const ENVIRONMENT_ALL = 'ENVIRONMENT_ALL';
 export const ENVIRONMENT_NOT_DEFINED = 'ENVIRONMENT_NOT_DEFINED';
```
Should these be renamed for clarity?
```suggestion
export const ENVIRONMENT_ALL_VALUE = 'ENVIRONMENT_ALL';
export const ENVIRONMENT_NOT_DEFINED_VALUE = 'ENVIRONMENT_NOT_DEFINED';
```

sqren replied:
thought: Perhaps the values and labels shouldn't be exported. Instead only the `ALL_OPTION` and `NOT_DEFINED_OPTION` are exported and then we do `ALL_OPTION.value` and `ALL_OPTION.label`?

sqren replied:
(not sure it actually makes a difference to readability though - just a thought)

sqren commented on x-pack/plugins/apm/public/components/shared/TransactionDurationAnomalyAlertTrigger/SelectAnomalySeverity.tsx:
```diff
@@ -45,24 +44,29 @@ const anomalyScoreSeverityMap: {
+      {label}
+    </EuiHealth>
+  );
+}
```
Nice with a dedicated component 👍

formgeist commented:
A possible bug: I noticed that the alerts popover doesn't close when the flyout is displayed.

<img width="1705" alt="Screenshot 2020-08-26 at 13 15 39" src="https://user-images.githubusercontent.com/4104278/91297271-511e2300-e79e-11ea-92d5-707b8a53338f.png">

I understand that because the anomaly detection job is on the environment, we have to write out the service name in the condition, but I also find it not useful for the end-user. The alerts are only currently available within the selected service view, so unless we're thinking of adding this alert type to the general APM view (no service selected), I think we can do with hiding it.

<img width="611" alt="Screenshot 2020-08-26 at 13 16 58" src="https://user-images.githubusercontent.com/4104278/91297480-b1ad6000-e79e-11ea-980e-a2bda72310cc.png">

_Example from duration threshold alert_

<img width="607" alt="Screenshot 2020-08-26 at 13 17 15" src="https://user-images.githubusercontent.com/4104278/91297483-b3772380-e79e-11ea-8be0-edeb8dd1778d.png">

_Existing duration anomaly alert_

sorenlouv reviewed (approved):
Overall this lgtm.
Would be good if we can find a better solution than the `DummyKibanaRequest` but not something blocking this PR

jgowdyelastic reviewed (approved):
ML changes LGTM

ogupte committed:
refactor get_all_environments to minimize exports and be more consistent and clean

ogupte committed:
- Reorg the alerts menu for different alert types (threshold/anomaly)
- default environment alert settings to the selected filter

ogupte committed:
- Filters default transaction type to only those supported in the APM anomaly detection jobs
- Removes Service name and transaction type from the set of expressions in the alerting setup

ogupte commented:
@formgeist, i reorganized the alerting menu to better support different alerting types for specific metrics:
![alerting-menu-reorg](https://user-images.githubusercontent.com/1967266/91369127-c6e6b680-e7bf-11ea-85b9-f70dac61ebe9.gif)

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/)
* Commit: b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/execution/node/654/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/execution/node/772/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/testReport)

<details><summary>X-Pack SAML API Integration Tests.x-pack/test/saml_api_integration/apis/security/saml_login·ts.apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/testReport//junit/X-Pack%20SAML%20API%20Integration%20Tests/x-pack_test_saml_api_integration_apis_security_saml_login·ts/apis_SAML_security_SAML_authentication_finishing_handshake_should_succeed_if_both_SAML_response_and_handshake_cookie_are_provided)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://dryrun

[00:00:00]       │
[00:00:00]         └-: apis SAML
[00:00:00]           └-> "before all" hook
[00:00:00]           └-: security
[00:00:00]             └-> "before all" hook
[00:00:00]             └-: SAML authentication
[00:00:00]               └-> "before all" hook
[00:00:00]               └-> should reject API requests if client is not authenticated
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] adding template [.kibana-event-log-8.0.0-template] for index patterns [.kibana-event-log-8.0.0-*]
[00:00:00]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] [.kibana-event-log-8.0.0-000001] creating index, cause [api], templates [.kibana-event-log-8.0.0-template], shards [1]/[1]
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] updating number_of_replicas to [0] for indices [.kibana-event-log-8.0.0-000001]
[00:00:00]                 └- ✓ pass  (232ms) "apis SAML security SAML authentication should reject API requests if client is not authenticated"
[00:00:00]               └-> does not prevent basic login
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ proc [kibana]   log   [01:59:47.468] [info][plugins][routes][security] Logging in with provider "basic" (basic)
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [.kibana-event-log-8.0.0-000001] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.kibana-event-log-8.0.0-000001][0]]])." previous.health="YELLOW" reason="shards started [[.kibana-event-log-8.0.0-000001][0]]"
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"new","action":"complete","name":"complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ proc [kibana]   log   [01:59:47.949] [warning][plugins][reporting] Enabling the Chromium sandbox provides an additional layer of protection.
[00:00:00]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] [ilm-history-2-000001] creating index, cause [api], templates [ilm-history], shards [1]/[0]
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [ilm-history-2-000001] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [ilm-history-ilm-policy]
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[ilm-history-2-000001][0]]])." previous.health="YELLOW" reason="shards started [[ilm-history-2-000001][0]]"
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [ilm-history-2-000001] from [{"phase":"new","action":"complete","name":"complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] in policy [ilm-history-ilm-policy]
[00:00:01]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [ilm-history-2-000001] from [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] in policy [ilm-history-ilm-policy]
[00:00:01]                 └- ✓ pass  (840ms) "apis SAML security SAML authentication does not prevent basic login"
[00:00:01]               └-: initiating handshake
[00:00:01]                 └-> "before all" hook
[00:00:03]               └-: finishing handshake
[00:00:03]                 └-> "before all" hook
[00:00:03]                 └-> should fail if SAML response is not complemented with handshake cookie
[00:00:03]                   └-> "before each" hook: global before each
[00:00:03]                   └-> "before each" hook
[00:00:03]                     │ proc [kibana]   log   [01:59:50.390] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:04]                   │ info [o.e.x.s.a.AuthenticationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] Authentication to realm saml1 failed - Provided SAML response is not valid for realm saml/saml1 (Caused by org.elasticsearch.ElasticsearchSecurityException: SAML content is in-response-to [_4f136e4b36ec153e70d264b35d6d289e3f6cce22] but expected one of [])
[00:00:04]                   └- ✓ pass  (227ms) "apis SAML security SAML authentication finishing handshake should fail if SAML response is not complemented with handshake cookie"
[00:00:04]                 └-> should succeed if both SAML response and handshake cookie are provided
[00:00:04]                   └-> "before each" hook: global before each
[00:00:04]                   └-> "before each" hook
[00:00:04]                     │ proc [kibana]   log   [01:59:51.524] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:05]                   │ info [o.e.x.s.s.SecurityIndexManager] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] security index does not exist. Creating [.security-tokens-7] with alias [.security-tokens]
[00:00:05]                   │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] [.security-tokens-7] creating index, cause [api], templates [], shards [1]/[0]
[00:00:05]                   │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.security-tokens-7][0]]])." previous.health="YELLOW" reason="shards started [[.security-tokens-7][0]]"
[00:00:08]                   └- ✖ fail: apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided
[00:00:08]                   │      Error: expected { username: 'a@b.c',
[00:00:08]                   │   roles: [],
[00:00:08]                   │   full_name: null,
[00:00:08]                   │   email: null,
[00:00:08]                   │   metadata: 
[00:00:08]                   │    { saml_nameid: 'a@b.c',
[00:00:08]                   │      saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified',
[00:00:08]                   │      'saml(urn:oid:0.0.7)': [ 'a@b.c' ] },
[00:00:08]                   │   enabled: true,
[00:00:08]                   │   authentication_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   lookup_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   authentication_type: 'token',
[00:00:08]                   │   authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
[00:00:08]                   │       at Assertion.assert (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:100:11)
[00:00:08]                   │       at Assertion.keys (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:505:8)
[00:00:08]                   │       at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
[00:00:08]                   │ 
[00:00:08]                   │ 

```

#### Stack Trace
```
Error: expected { username: 'a@b.c',
  roles: [],
  full_name: null,
  email: null,
  metadata: 
   { saml_nameid: 'a@b.c',
     saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified',
     'saml(urn:oid:0.0.7)': [ 'a@b.c' ] },
  enabled: true,
  authentication_realm: { name: 'saml1', type: 'saml' },
  lookup_realm: { name: 'saml1', type: 'saml' },
  authentication_type: 'token',
  authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
    at Assertion.assert (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.keys (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:505:8)
    at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
```
</details>

---

<details><summary>X-Pack SAML API Integration Tests.x-pack/test/saml_api_integration/apis/security/saml_login·ts.apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/testReport//junit/X-Pack%20SAML%20API%20Integration%20Tests/x-pack_test_saml_api_integration_apis_security_saml_login·ts/apis_SAML_security_SAML_authentication_finishing_handshake_should_succeed_if_both_SAML_response_and_handshake_cookie_are_provided_2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis SAML
[00:00:00]           └-> "before all" hook
[00:00:00]           └-: security
[00:00:00]             └-> "before all" hook
[00:00:00]             └-: SAML authentication
[00:00:00]               └-> "before all" hook
[00:00:00]               └-> should reject API requests if client is not authenticated
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] adding template [.kibana-event-log-8.0.0-template] for index patterns [.kibana-event-log-8.0.0-*]
[00:00:00]                 │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] [.kibana-event-log-8.0.0-000001] creating index, cause [api], templates [.kibana-event-log-8.0.0-template], shards [1]/[1]
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] updating number_of_replicas to [0] for indices [.kibana-event-log-8.0.0-000001]
[00:00:00]                 └- ✓ pass  (102ms) "apis SAML security SAML authentication should reject API requests if client is not authenticated"
[00:00:00]               │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [.kibana-event-log-8.0.0-000001] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [kibana-event-log-policy]
[00:00:00]               └-> does not prevent basic login
[00:00:00]                 └-> "before each" hook: global before each
[00:00:00]                 │ proc [kibana]   log   [01:24:11.549] [info][plugins][routes][security] Logging in with provider "basic" (basic)
[00:00:00]                 │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.kibana-event-log-8.0.0-000001][0]]])." previous.health="YELLOW" reason="shards started [[.kibana-event-log-8.0.0-000001][0]]"
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"new","action":"complete","name":"complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] in policy [kibana-event-log-policy]
[00:00:00]                 │ info [o.e.x.i.IndexLifecycleTransition] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] moving index [.kibana-event-log-8.0.0-000001] from [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] in policy [kibana-event-log-policy]
[00:00:01]                 └- ✓ pass  (914ms) "apis SAML security SAML authentication does not prevent basic login"
[00:00:01]               └-: initiating handshake
[00:00:01]                 └-> "before all" hook
[00:00:03]               └-: finishing handshake
[00:00:03]                 └-> "before all" hook
[00:00:03]                 └-> should fail if SAML response is not complemented with handshake cookie
[00:00:03]                   └-> "before each" hook: global before each
[00:00:03]                   └-> "before each" hook
[00:00:03]                     │ proc [kibana]   log   [01:24:14.569] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:04]                   │ info [o.e.x.s.a.AuthenticationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] Authentication to realm saml1 failed - Provided SAML response is not valid for realm saml/saml1 (Caused by org.elasticsearch.ElasticsearchSecurityException: SAML content is in-response-to [_016cc10471ee87eacaea8e4389098dabf03373a1] but expected one of [])
[00:00:04]                   └- ✓ pass  (192ms) "apis SAML security SAML authentication finishing handshake should fail if SAML response is not complemented with handshake cookie"
[00:00:04]                 └-> should succeed if both SAML response and handshake cookie are provided
[00:00:04]                   └-> "before each" hook: global before each
[00:00:04]                   └-> "before each" hook
[00:00:04]                     │ proc [kibana]   log   [01:24:15.637] [info][plugins][routes][security] Logging in with provider "saml" (saml)
[00:00:05]                   │ info [o.e.x.s.s.SecurityIndexManager] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] security index does not exist. Creating [.security-tokens-7] with alias [.security-tokens]
[00:00:05]                   │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] [.security-tokens-7] creating index, cause [api], templates [], shards [1]/[0]
[00:00:05]                   │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-centos-tests-xxl-1598487250642439569] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[.security-tokens-7][0]]])." previous.health="YELLOW" reason="shards started [[.security-tokens-7][0]]"
[00:00:08]                   └- ✖ fail: apis SAML security SAML authentication finishing handshake should succeed if both SAML response and handshake cookie are provided
[00:00:08]                   │      Error: expected { username: 'a@b.c',
[00:00:08]                   │   roles: [],
[00:00:08]                   │   full_name: null,
[00:00:08]                   │   email: null,
[00:00:08]                   │   metadata: 
[00:00:08]                   │    { saml_nameid: 'a@b.c',
[00:00:08]                   │      saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified',
[00:00:08]                   │      'saml(urn:oid:0.0.7)': [ 'a@b.c' ] },
[00:00:08]                   │   enabled: true,
[00:00:08]                   │   authentication_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   lookup_realm: { name: 'saml1', type: 'saml' },
[00:00:08]                   │   authentication_type: 'token',
[00:00:08]                   │   authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
[00:00:08]                   │       at Assertion.assert (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:100:11)
[00:00:08]                   │       at Assertion.keys (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:505:8)
[00:00:08]                   │       at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
[00:00:08]                   │ 
[00:00:08]                   │ 

```

#### Stack Trace
```
Error: expected { username: 'a@b.c',
  roles: [],
  full_name: null,
  email: null,
  metadata: 
   { saml_nameid: 'a@b.c',
     saml_nameid_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified',
     'saml(urn:oid:0.0.7)': [ 'a@b.c' ] },
  enabled: true,
  authentication_realm: { name: 'saml1', type: 'saml' },
  lookup_realm: { name: 'saml1', type: 'saml' },
  authentication_type: 'token',
  authentication_provider: 'saml' } to only have keys 'username', 'full_name', 'email', 'roles', 'metadata', 'enabled', 'authentication_realm', 'lookup_realm', and 'authentication_provider'
    at Assertion.assert (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:100:11)
    at Assertion.keys (/dev/shm/workspace/parallel/5/kibana/packages/kbn-expect/expect.js:505:8)
    at checkSessionCookie (test/saml_api_integration/apis/security/saml_login.ts:49:43)
```
</details>

---

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8) | diff | [baseline](https://github.com/elastic/kibana/commit/eecf4aa71f27b0fc22113f73ddf745a4fcd59cb5) |
| --- | --- | --- | --- |
| apm | 945 | +2 | 943 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8) | diff | [baseline](https://github.com/elastic/kibana/commit/eecf4aa71f27b0fc22113f73ddf745a4fcd59cb5) |
| --- | --- | --- | --- |
| apm | 4.7MB | +25.6KB | 4.7MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8) | diff | [baseline](https://github.com/elastic/kibana/commit/eecf4aa71f27b0fc22113f73ddf745a4fcd59cb5) |
| --- | --- | --- | --- |
| apm | 41.7KB | +1.1KB | 40.6KB |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8) | diff | [baseline](https://github.com/elastic/kibana/commit/eecf4aa71f27b0fc22113f73ddf745a4fcd59cb5) |
| --- | --- | --- | --- |
| total | 53204 | +9 | 53195 |

### History
* :broken_heart: [Build #70444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/) failed 94c3b48fb0f38921c0c0449c470c0adf52402ce2
* :green_heart: [Build #69862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/) succeeded ec08dcae074a283b6be7877fa177bf8ab2b66db9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/","number":"69862","commit":"ec08dcae074a283b6be7877fa177bf8ab2b66db9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/","number":"70444","commit":"94c3b48fb0f38921c0c0449c470c0adf52402ce2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/","number":"70742","commit":"b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8"}]}
PIPELINE-->

ogupte commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into apm-72636-anomaly-detection-based-alerts

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70762/)
* Commit: 936ae058927039916c8deba43cb07cf7f290a28e

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/936ae058927039916c8deba43cb07cf7f290a28e) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 945 | +2 | 943 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/936ae058927039916c8deba43cb07cf7f290a28e) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 4.7MB | +25.6KB | 4.7MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/936ae058927039916c8deba43cb07cf7f290a28e) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 41.7KB | +1.1KB | 40.6KB |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/936ae058927039916c8deba43cb07cf7f290a28e) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| total | 53244 | +9 | 53235 |

### History
* :broken_heart: [Build #70742](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/) failed b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8
* :broken_heart: [Build #70444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/) failed 94c3b48fb0f38921c0c0449c470c0adf52402ce2
* :green_heart: [Build #69862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/) succeeded ec08dcae074a283b6be7877fa177bf8ab2b66db9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/","number":"69862","commit":"ec08dcae074a283b6be7877fa177bf8ab2b66db9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/","number":"70444","commit":"94c3b48fb0f38921c0c0449c470c0adf52402ce2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/","number":"70742","commit":"b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70762/","number":"70762","commit":"936ae058927039916c8deba43cb07cf7f290a28e"}]}
PIPELINE-->

formgeist commented:
> @formgeist, i reorganized the alerting menu to better support different alerting types for specific metrics:
> ![alerting-menu-reorg](https://user-images.githubusercontent.com/1967266/91369127-c6e6b680-e7bf-11ea-85b9-f70dac61ebe9.gif)

Looks good, I think this makes sense. Minor nit; would you remove the icons from the "Create..." options inside the nested menu? They don't really belong there any longer.

ogupte commented:
@formgeist you suggested removing the service name read-only expression in the transaction duration anomaly alert setup. I also did the same for the transaction type since it will always be either `request` or `page-load` depending on the service. LMK if you think we should change it back:
![Screen Shot 2020-08-27 at 12 51 11 AM](https://user-images.githubusercontent.com/1967266/91413036-6aa38700-e7ff-11ea-9b85-7bf9682bda34.png)

ogupte committed:
- remove bell icon from alerts menu

formgeist commented:
@ogupte I had a think after our meeting yesterday, and this morning I put together some proposals for improvements. I don't want to unnecessarily block this PR, since what we have works. But in regards to the service name expression, I've suggested to keep it in order to allow for broader scoping of the alert. We discussed this yesterday briefly, but I've materialized the UX a bit more in my proposal https://github.com/elastic/kibana/issues/76068 Look forward to hear everyone's thoughts on these improvements.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70789/)
* Commit: 7ec51559fc09f572707979fd23b39ba42c231f74

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/7ec51559fc09f572707979fd23b39ba42c231f74) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 945 | +2 | 943 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/7ec51559fc09f572707979fd23b39ba42c231f74) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 4.7MB | +25.5KB | 4.7MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/7ec51559fc09f572707979fd23b39ba42c231f74) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 41.7KB | +1.1KB | 40.6KB |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/7ec51559fc09f572707979fd23b39ba42c231f74) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| total | 53244 | +9 | 53235 |

### History
* :green_heart: [Build #70762](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70762/) succeeded 936ae058927039916c8deba43cb07cf7f290a28e
* :broken_heart: [Build #70742](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/) failed b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8
* :broken_heart: [Build #70444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/) failed 94c3b48fb0f38921c0c0449c470c0adf52402ce2
* :green_heart: [Build #69862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/) succeeded ec08dcae074a283b6be7877fa177bf8ab2b66db9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/","number":"69862","commit":"ec08dcae074a283b6be7877fa177bf8ab2b66db9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/","number":"70444","commit":"94c3b48fb0f38921c0c0449c470c0adf52402ce2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/","number":"70742","commit":"b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70762/","number":"70762","commit":"936ae058927039916c8deba43cb07cf7f290a28e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70789/","number":"70789","commit":"7ec51559fc09f572707979fd23b39ba42c231f74"}]}
PIPELINE-->

ogupte committed:
Adds target service back into the anomaly alert setup as a ready-only expression

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71069/)
* Commit: 9d3a35a8bb3fb62d61bc23f125350c76b37841c0

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/9d3a35a8bb3fb62d61bc23f125350c76b37841c0) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 945 | +2 | 943 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/9d3a35a8bb3fb62d61bc23f125350c76b37841c0) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 4.7MB | +25.8KB | 4.7MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/9d3a35a8bb3fb62d61bc23f125350c76b37841c0) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| apm | 41.7KB | +1.1KB | 40.6KB |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/9d3a35a8bb3fb62d61bc23f125350c76b37841c0) | diff | [baseline](https://github.com/elastic/kibana/commit/a358c5768ea4f378270140d4e480fece98e8e103) |
| --- | --- | --- | --- |
| total | 53244 | +9 | 53235 |

### History
* :green_heart: [Build #70789](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70789/) succeeded 7ec51559fc09f572707979fd23b39ba42c231f74
* :green_heart: [Build #70762](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70762/) succeeded 936ae058927039916c8deba43cb07cf7f290a28e
* :broken_heart: [Build #70742](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/) failed b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8
* :broken_heart: [Build #70444](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/) failed 94c3b48fb0f38921c0c0449c470c0adf52402ce2
* :green_heart: [Build #69862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/) succeeded ec08dcae074a283b6be7877fa177bf8ab2b66db9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/69862/","number":"69862","commit":"ec08dcae074a283b6be7877fa177bf8ab2b66db9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70444/","number":"70444","commit":"94c3b48fb0f38921c0c0449c470c0adf52402ce2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70742/","number":"70742","commit":"b1f3b255b5a6aa7e49e024fd72f69daaa4a06bd8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70762/","number":"70762","commit":"936ae058927039916c8deba43cb07cf7f290a28e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/70789/","number":"70789","commit":"7ec51559fc09f572707979fd23b39ba42c231f74"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/71069/","number":"71069","commit":"9d3a35a8bb3fb62d61bc23f125350c76b37841c0"}]}
PIPELINE-->

cauemarcondes commented:
Tests ok
